Game/world/world.cpp

```diff
@@ -791,6 +795,20 @@ void World::addSound(const ZenLoad::zCVobData& vob) {
     }
   }
 
+bool World::addItemOrNpcBySymbolName (const std::string& name, const Tempest::Vec3& at) {
```

Comment:
This function is used only in single case (in marvin) - should be inlined, to keep World-class cleaner

Reply:
OK will do that, I am not really a friend of inline functions, but I can understand the argument :)

Reply:
I was thinking about why I dislike inline functions and variable initialization.
The reason is that I like to open the .h file to have a look at the interface of a class or object, and in most cases I do not care about the implementation details at that moment. So inline code will only make the interface harder to read in my opinion.
Also I think it will trigger a rebuild of all the files that include that interface if the inline function implementation is changed which would otherwise not be necessary.

But I am aware that this is probably my own personal preference and style so I have no problem with changing it if you pref ere the things to be inline :)

Reply:
While you might be right about the interface in VS there is CTRL+SHIFT+M for collapsing the Code and it looks clean again. Inlining functions - if the compiler decides to follow the request to inline - has the positive effect of removing the function prologue and epilogue from the assembly improving it for being integrated in performance relevant loops and for functions that are tiny and where the stack frame would nearly double the binary size of the function without inlining them. It adds to the size of the calling function however. So whether you like them is out of the question. It's a valid tool for a specific problem just like anything in C++, but in most cases it doesn't change that much for regular code. Good examples for inlining are math library member functions like vector math and stuff like that. They are called incredibly often in games so it does make a difference.

Reply:
Thanks for the detailed explanation, I only knew that it sometimes can improves compiler optimization.